pkg/client/client.go

```diff
@@ -256,6 +256,9 @@ func (c *Client) CreateOrUpdatePrometheus(p *monv1.Prometheus) error {
 	}
 
 	p.ResourceVersion = oldProm.ResourceVersion
+	if p.Spec.Storage != nil {
+		p.Spec.Storage.VolumeClaimTemplate.CreationTimestamp = metav1.Now()
```

Comment:
This means we're actually going to be updating every time as in fully modifying and bumping the `resourceVersion`, can we just set this to the 0 timestamp?

Reply:
I tried this, but this effectively becomes the zero value, being marshaled as `null`

Reply:
I'm just looking for any value that's static, so 1970-01-01T00:01Z works for me as well.

Reply:
isn't that a bit too hacky a fixed date? are we sure there's no side-effect on any of k/k controllers?

Reply:
judging by the output, the statefulset controller drops it

Reply:
@brancz doh, of course, let me update the PR.
@ant31 yes, this _is_ a hacky solution but gives us some air to fix prometheus operator.